sql_file3.sql

```diff
@@ -0,0 +1,705 @@
+-- 1 Extract the middle 3 characters from the string 'ABCDEFG'.
+SELECT SUBSTRING('ABCDEFG',3,3) AS MiddleCharacters;
+
+--2 From a table 'Employees' with a column 'FullName', write a query to extract the first name (assuming it's always the first word before a space).
+CREATE DATABASE TASK3;
+USE  TASK3;
+
+CREATE TABLE Employees(
+EmployeeID INT PRIMARY KEY,
+FullName VARCHAR(50));
+
+INSERT INTO  Employees VALUES
+(1,'Leo Das'),
+(2,'Mukhil M P'),
+(3,'Allen Jose');
+
+SELECT LEFT(FullName,CHARINDEX(' ',FullName) - 1) AS FirstName 
+FROM Employees;
+
+-- 3 Extract the first 5 characters from the string 'SQL Server 2022'.
+SELECT SUBSTRING('SQL Server 2022',1,5) AS FiveString;
+
+--4 From a 'Products' table with a 'ProductCode' column, write a query to get the first 3 characters of each product code.
+CREATE TABLE Products(
+ProductID CHAR(5) PRIMARY KEY);
+
+INSERT INTO Products VALUES
+('A1YZT'),
+('B2ORH'),
+('AK47S');
+
+SELECT LEFT(ProductID,3) AS StartThreeCharacters
+FROM Products;
+
+-- 5  Extract the last 4 characters from the string 'ABCDEFGHIJKL'
+
+SELECT RIGHT('ABCDEFGHIJKL',4) AS LastFourChar;
+
+-- 6 From an 'Orders' table with an 'OrderID' column (format: ORD-YYYY-NNNN), write a query to extract just the numeric portion at the end.
+CREATE TABLE Orders (
+    OrderID NVARCHAR(20)
+);
+
+INSERT INTO Orders (OrderID)
+VALUES 
+('ORD-2023-1234'),
+('ORD-2022-5678'),
+('ORD-2024-9101');
+
+SELECT OrderID, RIGHT(OrderID,4) AS Numeric
```

Comment:
How can we write this query generically, for example : if the ORDER ID is ORD-2023-123467 or ORD-2023-1234741.

Reply:
SELECT OrderID, RIGHT(OrderID, CHARINDEX('-', REVERSE(OrderID)) - 1) AS Numeric
FROM Orders;